src/BuildConfig.hpp

```diff
@@ -67,6 +72,13 @@ struct BuildConfig {
 public:
   explicit BuildConfig(const std::string& packageName, bool isDebug = true);
 
+  bool isExecutable() const {
+    return executable;
+  }
+  bool isLibrary() const {
+    return library;
+  }
```

Comment:
It's better to name `executable` as `isExecutable` but this conflicts with the function. To fix this problem, let me update the naming convention. This affects your code and probably produce conflicts. Apologies in advance.

Reply:
How about if I use underscores for now, like `_isExecutable`?

Reply:
Here are my thoughts:
* `_var` sounds like an unused variable.
* `var_` is confusing when it is a class instance, e.g., `var_.mem_`.
* `m_var` is weird since we are using `camelBack`.
* `mVar` can be confusing when using a single character identifier like `s` (`mS`).

So, I'm updating the naming convention to use `CamelCase` instead of `camelBack`, which is also used by [LLVM](https://llvm.org/docs/CodingStandards.html#name-types-functions-variables-and-enumerators-properly) that I am familiar with.

Reply:
By the way, if you have any idea, I'm happy to hear.

Reply:
So in the past, I've used `_varName` for private variables only. The reason being that you can use getters and setters like `type varName()` or `void varName(type val)`. That way you would also always avoid `var_.mem_` and instead it would become `var_.mem()` or even `var().mem()`.

Another idea is for getters and setters to always prefix it with `get`/`set` like `getVarName()`. So then my vars would become `isExecutable` and the getter would become `getIsExecutable()`.

Reply:
Thanks for your great insight.

The problem is that we don't always provide getters/setters since we have const members unlike Java:

```
struct SomeData {
    const int DataA;
    const int DataB;

    // We need provide getters for this simple struct...
};
```

Also, for big classes, providing corresponding getters/settings can make the code quite lengthy. The `isExecutable` variable is supposed to be read-only from outside, so providing `isExecutable` function makes sense.

What do you think?

Reply:
I personally don't see a problem with mixing getters and setters with public variables. Private variables exist for things that cannot under any circumstance be safely mutated or consumed without something "extra". So for that struct you showed, unless we needed to do something extra to `DataA` before the programmer is allowed to use it, we can just let them use it, no questions asked. Hopefully that makes sense. Ultimately my opinions aren't very strong on this and I'm willing to go with the flow here.

Reply:
I appreciate your input. I'll reconsider the naming convention as I started to think CamelCase isn't good either. Thanks!